polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -953,30 +993,11 @@ impl<Config: config::Config> XcmExecutor<Config> {
 			DepositReserveAsset { assets, dest, xcm } => {
 				let old_holding = self.holding.clone();
 				let result = Config::TransactionalProcessor::process(|| {
-					// we need to do this take/put cycle to solve wildcards and get exact assets to
-					// be weighed
-					let to_weigh = self.holding.saturating_take(assets.clone());
-					self.holding.subsume_assets(to_weigh.clone());
-					let to_weigh_reanchored = Self::reanchored(to_weigh, &dest, None);
-					let mut message_to_weigh =
-						vec![ReserveAssetDeposited(to_weigh_reanchored), ClearOrigin];
-					message_to_weigh.extend(xcm.0.clone().into_iter());
-					let (_, fee) =
-						validate_send::<Config::XcmSender>(dest.clone(), Xcm(message_to_weigh))?;
-					let maybe_delivery_fee = fee.get(0).map(|asset_needed_for_fees| {
-						tracing::trace!(
-							target: "xcm::DepositReserveAsset",
-							"Asset provided to pay for fees {:?}, asset required for delivery fees: {:?}",
-							self.asset_used_for_fees, asset_needed_for_fees,
-						);
-						let asset_to_pay_for_fees =
-							self.calculate_asset_for_delivery_fees(asset_needed_for_fees.clone());
-						// set aside fee to be charged by XcmSender
-						let delivery_fee =
-							self.holding.saturating_take(asset_to_pay_for_fees.into());
-						tracing::trace!(target: "xcm::DepositReserveAsset", ?delivery_fee);
-						delivery_fee
-					});
+					let maybe_delivery_fee_from_holding = if self.fees.is_empty() {
+						self.get_delivery_fee_from_holding(&assets, &dest, &xcm)?
+					} else {
+						None
+					};
```

Comment:
No because it only gets them from holding iff `self.fees.is_empty()`.

The `get_delivery_fee_from_holding()` fn is the "legacy" mechanism, while `self.fees` is the new one.

I like it how it is now because it keeps `get_delivery_fee_from_holding()` pure legacy, and we can remove it completely once deprecation period is done.